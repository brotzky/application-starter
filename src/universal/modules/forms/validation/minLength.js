export default function(min) {
  if (!min) return '';
  
  return function(value) {
    return value.length >= min
      ? undefined
      : {
          defaultMessage: `must be at least ${min} characters`,
          id: 'form.errors.minLength',
          values: { min: min },
        };
  }
}
