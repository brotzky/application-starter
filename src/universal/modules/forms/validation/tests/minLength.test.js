import minLength from '../minLength';

const atLeastThreeCharacters = minLength(3);

describe('[Form Validation] minLength', () => {
  describe('minLength()', () => {
    it('should return an empty string if nothing is passed it', () => {
      expect(minLength()).toMatchSnapshot();
    });

    it('should return a function', () => {
      expect(minLength(3)).toMatchSnapshot();
    });

    it('should return no errors if the number passed is more than the originally specified amount', () => {
      expect(atLeastThreeCharacters('1234')).toMatchSnapshot();
    });

    it('should return an error object when an amount less than specified is passed in', () => {
      expect(atLeastThreeCharacters('1')).toMatchSnapshot();
    });
  });
});
