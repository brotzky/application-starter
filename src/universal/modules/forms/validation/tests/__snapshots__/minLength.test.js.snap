// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Form Validation] minLength minLength() should return a function 1`] = `[Function]`;

exports[`[Form Validation] minLength minLength() should return an empty string if nothing is passed it 1`] = `""`;

exports[`[Form Validation] minLength minLength() should return an error object when an amount less than specified is passed in 1`] = `
Object {
  "defaultMessage": "must be at least 3 characters",
  "id": "form.errors.minLength",
  "values": Object {
    "min": 3,
  },
}
`;

exports[`[Form Validation] minLength minLength() should return no errors if the number passed is more than the originally specified amount 1`] = `undefined`;
